x-pack/plugins/fleet/server/services/epm/packages/install.ts

```diff
@@ -274,15 +278,12 @@ export async function handleInstallPackageFailure({
     createdAt: new Date().toISOString(),
     latestAttempts: installedPkg?.attributes.latest_install_failed_attempts,
   });
-
-  // if there is an unknown server error, uninstall any package assets or reinstall the previous version if update
+  // if there is an unknown server error, check the installType and do the following actions
   try {
     const installType = getInstallType({ pkgVersion, installedPkg });
-    if (installType === 'install') {
-      logger.error(`uninstalling ${pkgkey} after error installing: [${error.toString()}]`);
-      await removeInstallation({ savedObjectsClient, pkgName, pkgVersion, esClient });
-      return;
-    }
+    const attemptNumber = installedPkg?.attributes?.latest_install_failed_attempts?.length
```

Comment:
Are the `latest_install_failed_attempts` ever cleared? Maybe it should be cleared if a complete reinstall is happening, otherwise the retry logic will only happen once.

Reply:
yes that's already done, this was implemented a while ago.

Reply:
hm I see it cleared on a successful installation, though not cleared if the reinstall fails again, tested by throwing an error in `stepInstallIndexTemplatePipelines`

Reply:
I think that currently is cleared only when the installation is successful and keeps a maximum of 5 attempts. 

> I see it cleared on a successful installation, though not cleared if the reinstall fails again

I'm not sure why it should be cleared if the reinstall fails again? This way the reinstall can proceed with the retry flag.

Reply:
> I'm not sure why it should be cleared if the reinstall fails again? This way the reinstall can proceed with the retry flag.

I was thinking of a case where a reinstall failed with max retries because of a temporary reason (e.g. ES not responding for some time), and would be successful later. In that case we can probably just run the full reinstall again to get it working.